pymc3/step_methods/hmc/trajectory.py

```diff
@@ -57,6 +57,13 @@ def _theano_energy_function(H, q, **theano_kwargs):
     return energy_function, p
 
 
+def _theano_velocity_function(H, p, **theano_kwargs):
+    v = H.pot.velocity(p)
```

Comment:
Not sure what you mean. We need the velocity at the start of the trajectory. Later values are returned by leapfrog. This is pretty much the same as with the gradient.